Master/ Slave Inverter Modifications

AndyWhittaker opened the pull request:
Chris,

Here are the modifications I've made to your code to optionally allow the use of parallel inverters to work together.

Andy

celsworth commented:
Nice, thanks!

This looks good if you've never touched Ruby before :) My main concern is the master/slave code is always used even if you only have one inverter? So the MQTT messages will change etc, maybe need to make it optional and only change the messages if a slave is configured. I might be able to tidy it up to do that at some point. I'll leave this open to remind me as I don't have much time to work on this repo at the moment.

AndyWhittaker commented:
Hi Chris,

Glad you like it! The standard ini file headings have been kept the same to
"sort of'' maintain backward compatibility but, yes, I changed the MQTT
messages to masterinputs and added slaveinputs.

By the way, I developed the modifications using Visual Studio Code with
Ubuntu 20.04 running on Windows Subsystem for Linux. There is a
communication bug that keeps propping up and stopping the scanning, by the
way.

Andy

Andy Whittaker
ECUFix
Macclesfield
UK
g+: My Google+ <https://plus.google.com/+AndyWhittakerECUFix>
[image: Facebook] <https://www.facebook.com/ecufix.eu/>  [image: Blogger]
<http://blog.ecufix.com/>
Contact me: [image: Google Talk] mail@ecufix.com

This transmission, its content and any attachments are intended only for
the use of the recipient(s) and it may contain information that is
confidential and/or legally privileged.

If you are not the intended recipient you must not read, print, copy,
disseminate or use the communication for any purpose and should notify the
author by return e-mail.

While ECUFix uses Virus Checking software, it is strongly recommended that
you carry out your own virus check as we cannot accept liability for loss
or damage which may result directly or indirectly from this communication,
any attachments or hyperlinks.

ECUFix reserves the right to monitor and record e-mails addressed to
https://ecufix.com/


On Sun, 8 Nov 2020 at 18:04, Chris Elsworth <notifications@github.com>
wrote:

> Nice, thanks!
>
> This looks good if you've never touched Ruby before :) My main concern is
> the master/slave code is always used even if you only have one inverter? So
> the MQTT messages will change etc, maybe need to make it optional and only
> change the messages if a slave is configured. I might be able to tidy it up
> to do that at some point. I'll leave this open to remind me as I don't have
> much time to work on this repo at the moment.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/celsworth/octolux/pull/7#issuecomment-723645098>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGFTTS7ALX3CGARMPYEBWTSO3MSTANCNFSM4TONA2ZA>
> .
>
